lib/Sema/CSApply.cpp

```diff
@@ -7267,6 +7267,10 @@ Optional<SolutionApplicationTarget> ConstraintSystem::applySolution(
     if (!rewrittenExpr)
       return None;
 
+    if (target.getExprContextualTypePurpose() == CTP_Initialization) {
+
```

Comment:
This seems either incomplete or obsolete...

Reply:
Ah, it's incomplete. It's where the next bit of `typeCheckBinding` will be moving.